extension/include/boost/di/extension/injections/shared_factory.hpp

```diff
@@ -9,6 +9,10 @@
 #include <memory>
 
 #include "boost/di.hpp"
+#include "extensible_injector.hpp"
+#if BOOST_DI_THREAD_SAFE == true
```

Comment:
Wouldn't be better to have `BOOST_DI_THREAD_SAFE` defined or not? Instead of a boolean value? Being defined implies true, not defined implies false.

Reply:
But how can set it to true if not defined?